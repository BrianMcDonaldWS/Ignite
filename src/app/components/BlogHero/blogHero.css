.blogHero :global .title {
  padding-bottom: 3rem;
}
