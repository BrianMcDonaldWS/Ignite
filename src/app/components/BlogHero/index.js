import React from 'react';
import PropTypes from 'prop-types';
import makeClass from 'classnames';
import ReactRouterPropTypes from 'react-router-prop-types';

import styles from './blogHero.css';

const BlogHero = ({ location, blogHero }) => (
  <section
    className={makeClass('hero is-info is-medium is-bold', styles.blogHero)}
    style={
      blogHero && blogHero !== 'undefined'
        ? {
            maxWidth: 1800,
            margin: 'auto',
            background: `url(${blogHero})`,
            backgroundRepeat: 'no-repeat',
            backgroundSize: 'cover'
          }
        : {}
    }
  >
    <div className="hero-body">
      <div className="container has-text-centered">
        <h1
          className="title"
          style={
            location.pathname.includes('blog/index.md') ? {} : { opacity: 0 }
          }
        >
          Blog
        </h1>
      </div>
    </div>
  </section>
);

BlogHero.propTypes = {
  // eslint-disable-next-line react/no-typos
  location: ReactRouterPropTypes.location.isRequired,
  blogHero: PropTypes.string
};

BlogHero.defaultProps = {
  blogHero: null
};

export default BlogHero;
