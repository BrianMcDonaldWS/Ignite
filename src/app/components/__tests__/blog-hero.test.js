import React from 'react';

import BlogHero from '../BlogHero';
import renderToJson from './utils/render-to-json';

test('BlogHero - page - no hero', () => {
  expect(
    renderToJson(
      <BlogHero
        location={{ pathname: '/blog/page.md', hash: '', search: '' }}
      />
    )
  ).toMatchSnapshot();
});

test('BlogHero - page - hero', () => {
  expect(
    renderToJson(
      <BlogHero
        blogHero="https://some.link"
        location={{ pathname: '/blog/page.md', hash: '', search: '' }}
      />
    )
  ).toMatchSnapshot();
});

test('BlogHero - index has title', () => {
  expect(
    renderToJson(
      <BlogHero
        location={{ pathname: '/blog/index.md', hash: '', search: '' }}
      />
    )
  ).toMatchSnapshot();
});
