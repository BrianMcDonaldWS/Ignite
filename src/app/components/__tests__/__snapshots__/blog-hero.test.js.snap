// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogHero - index has title 1`] = `
<section
  className="hero is-info is-medium is-bold"
  style={null}
>
  <div
    className="hero-body"
  >
    <div
      className="container has-text-centered"
    >
      <h1
        className="title"
        style={Object {}}
      >
        Blog
      </h1>
    </div>
  </div>
</section>
`;

exports[`BlogHero - page - hero 1`] = `
<section
  className="hero is-info is-medium is-bold"
  style={
    Object {
      "background": "url(https://some.link)",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "cover",
      "margin": "auto",
      "maxWidth": 1800,
    }
  }
>
  <div
    className="hero-body"
  >
    <div
      className="container has-text-centered"
    >
      <h1
        className="title"
        style={
          Object {
            "opacity": 0,
          }
        }
      >
        Blog
      </h1>
    </div>
  </div>
</section>
`;

exports[`BlogHero - page - no hero 1`] = `
<section
  className="hero is-info is-medium is-bold"
  style={null}
>
  <div
    className="hero-body"
  >
    <div
      className="container has-text-centered"
    >
      <h1
        className="title"
        style={
          Object {
            "opacity": 0,
          }
        }
      >
        Blog
      </h1>
    </div>
  </div>
</section>
`;
